server/rest/src/main/java/org/infinispan/rest/stream/CacheChunkedStream.java

```diff
@@ -114,6 +116,17 @@ public void onError(Throwable t) {
             pendingBuffer = null;
          }
          cancel();
+         Channel.Unsafe unsafe;
+         if (ctx.channel() instanceof AbstractChannel) {
+            unsafe = ctx.channel().unsafe();
+         } else if (ctx.channel().parent() instanceof AbstractChannel) {
+            unsafe = ctx.channel().parent().unsafe();
+         } else {
+            ctx.close();
+            return;
+         }
+
+         unsafe.closeForcibly();
```

Comment:
Calling `ctx.close()` or the `closeForcibly()` directly from the channel calls the `AbstractHttp2StreamChannel.close(...) to send the `CANCEL` frame before closing the connection.

Reply:
So there is no difference between `ctx.close` and `unsafe.closeForcibly`?

Reply:
Maybe I am misreading, but it seems there is a difference that calling `unsafe.closeForcibly()` doesn't produce the cancel frame but calling `ctx.close()` would produce the frame right?

Reply:
Oh okay, the key difference is the parent then.

Reply:
Should we check the parent in the first if then instead? Or does the HTTP2 first if check not pass?

Reply:
Let me try that.

Reply:
I removed all the checks, and now it is just calling `ctx.close()` for simplicity. When using HTTP/1, all is good. If it is HTTP/2, it sends the cancel the frame first, but it ends up closing the connection after.